src/site/tools/pub/publishing.markdown

```diff
@@ -7,29 +7,89 @@ title: "Publishing a Package"
 
 # {{ page.title }}
 
-[Pub](/tools/pub) isn't just for using other people's packages. It also allows you to share
-your packages with the world. Once you've written some useful code and you want
-everyone else to be able to use it, just run:
+[Pub](/tools/pub) isn't just for using other people's packages.
+It also allows you to share your packages with the world. If you have a useful
+project and you want others to be able to use it, use the `pub publish`
+command. For your first time around, you can perform a dry run like so:
+
+{% prettify sh %}
+$ pub publish --dry-run
+{% endprettify %}
+
+Pub will check to make sure that your package follows the
+[pubspec format](pubspec.html) and
+[package layout conventions](package-layout.html), and then upload your package
+to [pub.dartlang.org](https://pub.dartlang.org). Pub will also show you all of
+the files it intends to "publish". Here's an example of publishing a package
+named `transmogrify`:
+
+{% prettify none %}
+Publishing transmogrify 1.0.0 to https://pub.dartlang.org:
```

Comment:
So, I think developers will be coming from other languages where you specify a glob or a list of files that will go into your package ([Maven, I think](http://maven.apache.org/pom.html#Resources), [Rubygems](http://guides.rubygems.org/specification-reference/#files), [npm's package.json](https://docs.npmjs.com/files/package.json#files).

To me, I think it's important to see the files that pub is going to publish to the world. And in fact, this is essentially the only output that `pub publish` prints (also warnings and a "are you sure?")

I doubt this output changes much at all because it is literally this one line ("Publishing <name> <version> to <url>:") and then the tree of files.